src/FunctionApps/BulkImport/Import/main.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import os
+import zipfile
+
+import requests
+import multiprocessing
+import json
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+unzipped_directory = "./FhirResources"
+
+def main(zip_file):
+    unzip_input_file(zip_file)
+    process_ndjson_files(unzipped_directory)
+
+def unzip_input_file(zip_file):
+    with zipfile.ZipFile(zip_file,"r") as zip_ref:
+        zip_ref.extractall(unzipped_directory)
+
+def process_ndjson_files(dir):
+    with multiprocessing.Pool(processes = multiprocessing.cpu_count()-1) as p:
+        directory = os.fsencode(dir)
+        for file in os.listdir(directory):
+            filename = os.fsdecode(file)
+            if filename.endswith(".ndjson"):
+                file_path = (os.path.join(dir, filename))
+                p.apply_async(read_file, (file_path,))
+            else:
+                logging.error(f"File must be ndjson, file given: {filename}")
+        p.close()
+        p.join()
+
+def read_file(file):
+    with open(file) as fp:
+        for line in fp:
+            json_line = json.loads(line)
+            resource_type = json_line["resourceType"]
+            resource_id   = json_line["id"]
+            post_to_fhir(line,resource_type,resource_id)
+
+def post_to_fhir(line,resource_type,resource_id):
+    try:
+        token = get_access_token()
+    except Exception:
+        return func.HttpResponse("error getting access token", status_code=401) 
+    retry_strategy = Retry(
+    total=3,
+    status_forcelist=[429, 500, 502, 503, 504],
+    allowed_methods=["HEAD", "POST", "OPTIONS"]
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
```

Comment:
Is this an access token for the FHIR server? Or to be able to run the function in the environment?

Reply:
This is the access token, once we orchestrate things will change as we have the new auth method.